pallets/schemas/src/lib.rs

```diff
@@ -78,40 +79,89 @@ pub mod pallet {
 	#[pallet::generate_store(pub(super) trait Store)]
 	pub struct Pallet<T>(_);
 
+	#[pallet::storage]
+	#[pallet::getter(fn get_schema_max_bytes)]
+	pub(super) type GovernanceSchemaMaxBytes<T: Config> = StorageValue<_, u32, ValueQuery>;
+
 	#[pallet::storage]
 	#[pallet::getter(fn schema_count)]
 	pub(super) type SchemaCount<T: Config> = StorageValue<_, SchemaId, ValueQuery>;
 
 	// storage for message schemas hashes
 	#[pallet::storage]
 	#[pallet::getter(fn get_schema)]
-	pub(super) type Schemas<T: Config> =
-		StorageMap<_, Twox64Concat, SchemaId, BoundedVec<u8, T::MaxSchemaSizeBytes>, OptionQuery>;
+	pub(super) type Schemas<T: Config> = StorageMap<
+		_,
+		Twox64Concat,
+		SchemaId,
+		BoundedVec<u8, T::SchemaMaxBytesBoundedVecLimit>,
+		OptionQuery,
+	>;
+
+	#[pallet::genesis_config]
+	pub struct GenesisConfig {
+		pub initial_max_schema_size: u32,
+	}
+
+	#[cfg(feature = "std")]
+	impl sp_std::default::Default for GenesisConfig {
+		fn default() -> Self {
+			Self { initial_max_schema_size: 1024 }
+		}
+	}
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig {
+		fn build(&self) {
+			GovernanceSchemaMaxBytes::<T>::put(self.initial_max_schema_size.clone());
+		}
+	}
 
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
-		#[pallet::weight(<T as Config>::WeightInfo::register_schema(schema.len() as u32))]
-		pub fn register_schema(origin: OriginFor<T>, schema: Vec<u8>) -> DispatchResult {
+		#[pallet::weight(< T as Config >::WeightInfo::register_schema(schema.len() as u32))]
+		pub fn register_schema(
+			origin: OriginFor<T>,
+			schema: BoundedVec<u8, T::SchemaMaxBytesBoundedVecLimit>,
+		) -> DispatchResult {
 			let sender = ensure_signed(origin.clone())?;
 
+			ensure!(
+				schema.len() > T::MinSchemaSizeBytes::get() as usize,
+				Error::<T>::LessThanMinSchemaBytes
+			);
+			ensure!(
+				schema.len() < Self::get_schema_max_bytes() as usize,
+				Error::<T>::ExceedsMaxSchemaBytes
+			);
+
 			let cur_count = Self::schema_count();
-			ensure!(cur_count < T::MaxSchemaRegistrations::get(), <Error<T>>::TooManySchemas);
-			let schema_id = cur_count.checked_add(1).ok_or(<Error<T>>::SchemaCountOverflow)?;
+			ensure!(cur_count < T::MaxSchemaRegistrations::get(), Error::<T>::TooManySchemas);
+			let schema_id = cur_count.checked_add(1).ok_or(Error::<T>::SchemaCountOverflow)?;
 
 			Self::add_schema(schema.clone(), schema_id)?;
 
 			Self::deposit_event(Event::SchemaRegistered(sender, schema_id));
 			Ok(())
 		}
+
+		/// Set a new value for the Schema maximum number of bytes.  Must be <= the limit of the
+		/// Schema BoundedVec used for registration.
+		#[pallet::weight(30_000)]
+		pub fn set_max_schema_bytes(origin: OriginFor<T>, max_size: u32) -> DispatchResult {
+			ensure_root(origin.clone())?;
+			ensure!(max_size <= T::SchemaMaxBytesBoundedVecLimit, Error::<T>::InvalidSchemaMaxValue);
+			GovernanceSchemaMaxBytes::<T>::set(max_size);
```

Comment:
I think we do need events for any governance related changes

Reply:
Argh, you're right!